System/Metrics/Json.hs

```diff
@@ -61,6 +69,35 @@ sampleToJson metrics =
         Just m' -> A.Object $ M.insert str (go m' rest val) m
     go v _ _                        = typeMismatch "Object" v
 
+    -- | Get the formatted labels for the metric
+    --
+    -- Replace the escaped sequence ("\\.") with the replacment("\\_dot_\\")
```

Comment:
@Naveenaidu why not replace `\\.` with `_dot_` ? instead of `\\_dot_\\`? Just simplifying things. Also what happens if metric label contains the word or substring `dot` (e.g `somefoo.ancedote.bar` or `somefoo.dot.bar` etc.)?  Have you tried that out?

Reply:
This was one thing which was bugging me. The reason I chose to go with `\\_dot_\\` was that there might be chances of having a `_dot_` in a label name and the chances of having `\\_dot_\\` was less. But you are right, I should not put my faith in the probability of the string not containing `\\_dot_\\` , because if on the off chance the string ends up containing this, then we will be mutating the string. And that we have to avoid.

Reply:
We decided to use the [split function](https://hackage.haskell.org/package/pcre-heavy-1.0.0.2/docs/src/Text-Regex-PCRE-Heavy.html#split) for the [Text.Regex.PCRE.Heavy](https://hackage.haskell.org/package/pcre-heavy-1.0.0.2/docs/Text-Regex-PCRE-Heavy.html) and use a negative look-behind regex to match only those dots `.` which are not prepended by `\\`